lib/hathifiles_database/db/connection.rb

```diff
@@ -16,27 +16,36 @@ module HathifilesDatabase
   class DB
     class Connection
       extend HathifilesDatabase::Exception
-
-      LOGGER = Logger.new($stderr)
       MIGRATION_DIR = Pathname.new(__dir__) + "migrations"
 
       attr_accessor :logger, :rawdb, :slice_size
 
       # We take the name of the main table from the constant
       # MAINTABLE and the names of the foreign tables from the
       # keys in the line's #foreign_table_data hash
-      # @param [String] connection_string A valid Sequel connection string
-      #   (see https://sequel.jeremyevans.net/rdoc/files/doc/opening_databases_rdoc.html)
-      # @param [#info] logger A logger object that responds to, e.g., `#warn`,
+
+      # @param [Hash] kwargs
+      # @option kwargs [String] :username (database connection)
+      # @option kwargs [String] :password (database connection)
+      # @option kwargs [String] :host (database connection)
+      # @option kwargs [String] :database (database connection)
+      # @option kwargs [Logger] :logger object that responds to, e.g., `#warn`,
       #   `#info`, etc.
-      def initialize(connection_string, logger: LOGGER)
-        @rawdb = Sequel.connect(connection_string + "?local_infile=1&CharSet=utf8mb4")
-        # __setobj__(@rawdb)
+      def initialize(**kwargs)
+        @rawdb = Sequel.connect(
+          adapter: "mysql2",
+          user: kwargs[:username] || ENV["MARIADB_HATHIFILES_RW_USERNAME"],
```

Comment:
Regarding overriding I think this is correct. I'm not sure _when_ we'd end up using the kwargs version (except maybe for some kinds of testing) but I don't think this is a problem.